mobilecoind-json/src/main.rs

```diff
@@ -312,12 +312,12 @@ fn transfer(
         hex::decode(monitor_hex).map_err(|err| format!("Failed to decode monitor hex: {}", err))?;
 
     // Decode the keys
-    let mut view_public_key = RistrettoPublic::new();
+    let mut view_public_key = CompressedRistretto::new();
     view_public_key.set_data(
         hex::decode(&transfer.receiver.view_public_key)
             .map_err(|err| format!("Failed to decode view key hex: {}", err))?,
     );
-    let mut spend_public_key = RistrettoPublic::new();
+    let mut spend_public_key = CompressedRistretto::new();
```

Comment:
Can these use the `From` traits?

Reply:
@eranrund so actually I guess i'm confused, it looks like these are protobuf CompressedRistretto but they are using a hex encoding, I'm not sure why.

Is it okay to remove the hex encoding? I think it will break mobilecoind api, right? CompressedRistretto in external.proto is not normally hex encoded in this way

Reply:
Can you link to what you're referring?

Reply:
in the lines immediately below this line you commented on, it looks like

```
    // Decode the keys
    let mut view_public_key = CompressedRistretto::new();
    view_public_key.set_data(
        hex::decode(&transfer.receiver.view_public_key)
            .map_err(|err| format!("Failed to decode view key hex: {}", err))?,
    );
    let mut spend_public_key = CompressedRistretto::new();
    spend_public_key.set_data(
        hex::decode(&transfer.receiver.spend_public_key)
            .map_err(|err| format!("Failed to decode spend key hex: {}", err))?,
    );
```

If I do `external::CompressedRistretto::from(&view_public_key)` instead of this code, I think that will be a behavior change

Reply:
maybe i'm confused, I guess receiver .view_public_key is not a `mc_crypto_keys` object?

Reply:
You're correct, it is not. I think this can stay as it, my bad.

Reply:
This code is from the other PR, unfortunately because both PRs are in my fork and not this repo, I can't stage this PR over that one, they both have to point at master, which makes it a little harder to review

I moved the code in question into a conversion anyways, lmk what you think in the other PR